pkg/cluster/cluster.go

```diff
@@ -82,10 +82,11 @@ type Cluster struct {
 	currentProcess   Process
 	processMu        sync.RWMutex // protects the current operation for reporting, no need to hold the master mutex
 	specMu           sync.RWMutex // protects the spec for reporting, no need to hold the master mutex
+	dryRunMode       bool
```

Comment:
Why is this a property of a cluster? Do we expect to run certain clusters in a dry-run mode but not others (wouldn't it be error prone to do so?). If not, I'd put it to the controller and consult this flag when proceeding to execute specific actions (update, delete, sync cluster).